src/DynamicData.Tests/Binding/AvaloniaDictionaryFixture.cs

```diff
@@ -0,0 +1,303 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+
+using  DynamicData.Binding;
+using DynamicData.Tests.Domain;
+using FluentAssertions;
+using Xunit;
+
+namespace DynamicData.Tests.Binding;
+
+public class AvaloniaDictionaryFixture
+{
+    private readonly AvaloniaDictionary<string, Person> _collection;
+    private readonly ChangeSetAggregator<Person> _results;
+
+    public AvaloniaDictionaryFixture()
+    {
+        _collection = new AvaloniaDictionary<string, Person>();
+        _results =  _collection.ToObservableChangeSet<AvaloniaDictionary<string, Person>, KeyValuePair<string, Person>>()
+            .Transform(x=>x.Value)
+            .AsAggregator();
+    }
+
+    [Fact]
+    public void Add()
+    {
+        var person = new Person("Someone",10, "M");
+
+        _collection.Add("Someone", person);
+
+        _results.Messages.Count.Should().Be(1);
+        _results.Data.Count.Should().Be(1);
+        _results.Data.Items.First().Should().Be(person);
+    }
+
+    [Fact]
+    public void Replace()
```

Comment:
This is the particular scenario that was causing exceptions, right?

Reply:
In fact each of the methods in this test causes an exception.